### PR TITLE
Save walkthrough state on mobile

### PR DESCRIPTION
**Pull request by lubej**
3 step walkthrough for ecosystem shows up repeatedly on mobile. Save state.

### CODE REVIEW

File: src/app/pages/HomePage/Graph/para-time-selector-utils.ts [hunk 1]
```diff
@@ -11,7 +15,9 @@ export abstract class ParaTimeSelectorUtils {
   }
 
   static showMobileHelpScreen(step: ParaTimeSelectorStep, isMobile: boolean) {
```
Comment (lukaw3d) [inline src/app/pages/HomePage/Graph/para-time-selector-utils.ts:17]: "Toggle info screen" doesn't work anymore

File: src/app/utils/storage.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { AppError, AppErrors } from '../../types/errors'
+import { StorageKeys } from '../../types/storage'
+
+export const storage = (storage = localStorage) => {
+  const set = <T = unknown>(key: StorageKeys, value: T): void => {
+    try {
+      const serializedValue = JSON.stringify(value)
+      storage.setItem(key, serializedValue)
```
Comment (lukaw3d) [inline src/app/utils/storage.ts:8]: Nice!

File: src/app/utils/storage.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    } catch (error) {
+      throw new AppError(AppErrors.Storage)
+    }
+  }
+
+  const get = <T = unknown>(key: StorageKeys): T | undefined => {
```
Comment (lukaw3d) [inline src/app/utils/storage.ts:14]: We can predefine our stored value types too, instead of T = unknown

File: src/locales/en/translation.json [hunk 4]
```diff
@@ -114,7 +114,8 @@
     "notFoundTx": "Transaction not found",
     "validateURL": "Please validate provided URL",
     "validateURLOrGoToFirstTab": "Please check the URL or load the first tab.",
-    "invalidUrl": "Invalid URL"
+    "invalidUrl": "Invalid URL",
+    "storage": "Access to storage denied"
```
Comment (lukaw3d) [inline src/locales/en/translation.json:118]: I think this should mention browser storage

File: src/app/pages/HomePage/Graph/ParaTimeSelector/index.tsx [hunk 5]
```diff
@@ -161,7 +166,7 @@ const ParaTimeSelectorCmp: FC<ParaTimeSelectorProps> = ({ disabled, step, setSte
   }, [isMobile, step, setStep])
 
   const onExploreClick = () => {
-    if (isMobile) {
+    if (isMobile && !mobileHelpScreenShown) {
```
Comment (lukaw3d) [inline src/app/pages/HomePage/Graph/ParaTimeSelector/index.tsx:169]: If user clicks Explore Oasis, goes through the help, clicks Emerald, clicks logo in topbar, clicks Explore Oasis again: then graph stops working

Comment (lubej) [reply src/app/pages/HomePage/Graph/ParaTimeSelector/index.tsx:169]: Resolved 👍

File: src/app/pages/HomePage/Graph/para-time-selector-utils.ts (hunk 1 above)
Comment (lubej) [reply src/app/pages/HomePage/Graph/para-time-selector-utils.ts:17]: Nice catch. Resolved 👍

File: src/app/pages/HomePage/Graph/ParaTimeSelector/index.tsx [hunk 6]
```diff
@@ -161,7 +166,9 @@ const ParaTimeSelectorCmp: FC<ParaTimeSelectorProps> = ({ disabled, step, setSte
   }, [isMobile, step, setStep])
 
   const onExploreClick = () => {
-    if (isMobile) {
+    const mobileHelpScreenShown = localStore.get(StorageKeys.MobileHelpScreenShown)
+
+    if (isMobile && !mobileHelpScreenShown) {
```
Comment (lukaw3d) [inline src/app/pages/HomePage/Graph/ParaTimeSelector/index.tsx:171]: It still breaks after:
- user clicks Explore Oasis
- goes through the help
- clicks Toggle info screen

or
- refreshes
- clicks Toggle info screen
- goes through the help
- clicks Toggle info screen

File: src/app/utils/storage.ts (hunk 3 above)
Comment (lukaw3d) [reply src/app/utils/storage.ts]: > ```ts
> type StorageType = string
> const get = <T = StorageType>(key: StorageKeys): T | undefined => {
> ```
not like that

Comment (lukaw3d) [reply src/app/utils/storage.ts]: Like this
```ts
interface StorageType {
  'mobile-help-screen-shown': boolean
}
const get = <K extends keyof StorageType>(key: K): StorageType[K] | undefined => {
```

File: src/app/pages/HomePage/Graph/ParaTimeSelector/index.tsx (hunk 6 above)
Comment (lubej) [reply src/app/pages/HomePage/Graph/ParaTimeSelector/index.tsx:171]: Resolved now, minor bug. Thanks for noticing yet again

**Review by lukaw3d — APPROVED** [review]

**Pull request merged**